ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/service/internal/EBINuFileService.java

```diff
@@ -0,0 +1,194 @@
+import eu.elixir.ega.ebi.dataedge.exception.UnretrievableFileException;
+import eu.elixir.ega.ebi.dataedge.service.FileMetaService;
+import eu.elixir.ega.ebi.dataedge.service.KeyService;
+import eu.elixir.ega.ebi.dataedge.service.NuFileService;
+import eu.elixir.ega.ebi.dataedge.utils.DecryptionUtils;
+import htsjdk.samtools.seekablestream.cipher.ebi.Glue;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.BoundedInputStream;
+import org.springframework.security.core.Authentication;
+import uk.ac.ebi.ega.fire.exceptions.ClientProtocolException;
+import uk.ac.ebi.ega.fire.exceptions.FireServiceException;
+import uk.ac.ebi.ega.fire.service.IFireService;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.TimeUnit;
+
+public class EBINuFileService implements NuFileService {
+
+    private final KeyService keyService;
+    private final FileMetaService fileMetaService;
+    private final IFireService fireClientService;
+    private final LoadingCache<String, byte[]> aesHeaderCache;
+
+    public EBINuFileService(KeyService keyService, FileMetaService fileMetaService, IFireService fireClientService) {
+        this.keyService = keyService;
+        this.fileMetaService = fileMetaService;
+        this.fireClientService = fireClientService;
+        this.aesHeaderCache = Caffeine.newBuilder()
+                .expireAfterAccess(1, TimeUnit.DAYS)
+                .build(this::getAESHeader);
+    }
+
+    @Override
+    public long getPlainFileSize(String fileId, Authentication auth, String sessionId) throws EgaFileNotFoundException, UnretrievableFileException, FileNotAvailableException {
+
+        return getPlainFileSize(getFile(fileId, auth, sessionId), keyService.getEncryptionAlgorithm(fileId));
+    }
+
+    private File getFile(String fileId, Authentication auth, String sessionId) throws EgaFileNotFoundException, FileNotAvailableException {
+        File file;
+        try {
+            file = fileMetaService.getFile(auth, fileId, sessionId);
+        } catch(NotFoundException exception) {
+            throw new EgaFileNotFoundException(fileId);
+        }
+        if (file.getFileId() == null) {
+            throw new FileNotAvailableException(fileId);
+        }
+        return file;
+    }
+
+    private long getPlainFileSize(File file, String encryptionFormat) throws UnretrievableFileException {
+        long size = file.getFileSize();
+
+        if (encryptionFormat.equals("aes128") || encryptionFormat.equals("aes256")) {
```

Comment:
You can write above condition as `"aes128".equals(encryptionFormat ) || "aes256".equals(encryptionFormat )`. Make such a change wherever needed.